anyway/backend_constants.py

```diff
@@ -120,3 +85,111 @@ class ResolutionCategories(Enum):
 
 
 BE_CONST = BackEndConstants()
+
+
+class LabeledCode(Enum):
+    @staticmethod
+    def get_label_impl(cls, val: int, labels: List[str]) -> str:
+        if 0 <= val - 1 < len(labels):
+            return labels[val - 1]
+        else:
+            err = f'{val}: not found in class {cls}'
+            logging.error(err)
+            raise ValueError(err)
+
+    @classmethod
+    def codes(cls: Iterable) -> List[int]:
+        if isinstance(cls, collections.Iterable):
+            # pylint: disable=E1133
+            return [a.value for a in cls]
+        else:
+            raise ValueError(f'{cls}: needs to be derived from Enum')
+
+    # def get_label(self) -> str:
```

Comment:
Why is this function commented out?

Reply:
It is wrong. Enum items are not instances of the class, but rather static attributes. I will remove it.

Reply:
I'm not sure I understand.
Enum instances are instances of the class (_and_ static attributes).

Reply:
The wrong thing about this function is the use of `labels` as a static fiels; Enums cannot have static fields for obvious reasons, so you need a `staticmethod`.
```python
from enum import Enum

class A(Enum):
    X = 0
    Y = 1
    
    def get_label(self):
        return A.labels()[self.value]
    
    @staticmethod
    def labels():
        return ["I am X", "I am Y"]
    
assert(isinstance(A.X, A))  # assertion succeeds
print(A.X.get_label())   # prints "I am X"
```

Reply:
Another option would be to use [the special `_ignore_` field](https://docs.python.org/3/library/enum.html#allowed-members-and-attributes-of-enumerations), so we can give a type to a field that is monkey-patched later:
```python
from typing import List
from enum import Enum

class A(Enum):
    _ignore_ = ['labels']
    labels: List[str]
    
    X = 0
    Y = 1
    
    def get_label(self):
        # `A.labels` would work in this specific example, but we probably want `type(self).labels`
        # Or we can take advantage of the fact that you can access static fields as if they are members, and just use `self`:
        return self.labels[self.value]
    
# Monkey-patch the enum to have a non-enum static member
A.labels = ["I am X", "I am Y"]
    
assert(isinstance(A.X, A))  # assertion succeeds
print(A.X.get_label())   # prints "I am X"
```
See also [this SO answer](https://stackoverflow.com/a/56854002/2289509).

Reply:
Eventually I used a similar solution. As the table is only needed in one method, define it on-the-fly in that method.
In addition, I need a class that provides the additional methods.